commands/src/main/java/org/wildfly/extras/creaper/commands/socketbindings/AddSocketBinding.java

```diff
@@ -0,0 +1,174 @@
+package org.wildfly.extras.creaper.commands.socketbindings;
+
+import org.wildfly.extras.creaper.core.CommandFailedException;
+import org.wildfly.extras.creaper.core.online.OnlineCommand;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * <p>A command for creating socket binding under default socket binding group if not specified else.</p>
+ * <p>This command <b>does not check</b> whether the port or multicast port are available!</p>
+ *
+ * @author Jan Kasik <jkasik@redhat.com>
+ */
+public class AddSocketBinding implements OnlineCommand {
+
+    private final String socketBindingName;
+    private final String socketBindingGroup;
+
+    private final List<String> clientMappings;
+    private final Boolean fixedPort;
+    private final String interfaceName;
+    private final String multicastAddress;
+    private final Integer multicastPort;
+    private final Integer port;
+    private final boolean replaceExisting;
+
+    private AddSocketBinding(Builder builder) {
+        this.socketBindingName = builder.socketBindingName;
+        this.socketBindingGroup = builder.socketBindingGroup;
+        this.clientMappings = builder.clientMappings;
+        this.fixedPort = builder.fixedPort;
+        this.interfaceName = builder.interfaceName;
+        this.multicastAddress = builder.multicastAddress;
+        this.multicastPort = builder.multicastPort;
+        this.port = builder.port;
+        this.replaceExisting = builder.replaceExisting;
+    }
+
+    @Override
+    public void apply(OnlineCommandContext ctx) throws CommandFailedException, IOException {
+        Operations ops = new Operations(ctx.client);
+        String contextSocketBindingGroup;
+
+        if (socketBindingGroup == null) {
+            contextSocketBindingGroup = ctx.client.options().isDomain ? "full-sockets" : "standard-sockets";
+        } else {
+            contextSocketBindingGroup = socketBindingGroup;
+        }
+
```

Comment:
I think I would prefer something like this to inventing a new name `context...`:

```
String socketBindingGroup = this.socketBindingGroup;
if (socketBindingGroup == null) {
    ...
}
```

Also, your choice of default socket binding group needs a bit of explanation. It's trivial for standalone, but not so much for managed domain. I guess that `full-sockets` is chosen because it's the socket binding group used by all the servers that are started in the default domain configuration (i.e., `server-one` and `server-two` from `main-server-group`; there's `server-three` from `other-server-group`, but that's not started by default). It kinda makes sense, but should probably be mentioned in the class javadoc.